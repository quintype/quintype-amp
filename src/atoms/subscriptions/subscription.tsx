import React, { Fragment } from "react";
import { Helmet } from "react-helmet";
import { SubscriptionProps } from "./types";

export const Subscription = ({ services, score, fallbackEntitlement, story }: SubscriptionProps) => {
  const ampSubscriptionProps = {
    ...(services && { services }),
    ...(score && { score }),
    ...(fallbackEntitlement && { fallbackEntitlement })
  };
  const isSubscribed = story.access === "subscription";
  return (
    <Fragment>
      <Helmet>
        {isSubscribed && (
          <script
            async={undefined}
            custom-element="amp-subscriptions"
            src="https://cdn.ampproject.org/v0/amp-subscriptions-0.1.js"
          />
        )}
      </Helmet>
      {isSubscribed && (
        <script
          type="application/json"
          id="amp-subscriptions"
          dangerouslySetInnerHTML={{ __html: JSON.stringify(ampSubscriptionProps) }}
        />
      )}
    </Fragment>
  );
};
