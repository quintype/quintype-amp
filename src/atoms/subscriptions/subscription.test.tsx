import { shallow } from "enzyme";
import React from "react";
import { Subscription } from "./subscription";

const services = [
  {
    authorizationUrl: "https://pub.com/amp-authorisation?rid=READER_ID&url=SOURCE_URL",
    pingbackUrl: "https://pub.com/amp-pingback?rid=READER_ID&url=SOURCE_URL",
    actions: {
      login: "https://pub.com/amp-login?rid=READER_ID&url=SOURCE_URL",
      subscribe: "https://pub.com/amp-subscribe?rid=READER_ID&url=SOURCE_URL"
    }
  }
];
const noServices = [
  {
    authorizationUrl: "",
    pingbackUrl: "",
    actions: {
      login: "",
      subscribe: ""
    }
  }
];
const score = {
  supportsViewer: 10,
  isReadyToPay: 9
};
const fallbackEntitlement = {
  source: "fallback",
  granted: true,
  grantReason: "SUBSCRIBER",
  isLoggedIn: false
};
describe("Subscriptions", () => {
  it("should render Subscription services ", () => {
    const wrapper = shallow(<Subscription services={services} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should render Subscription services with score ", () => {
    const wrapper = shallow(<Subscription services={services} score={score} />);
    expect(wrapper.find("script[type='application/json']").html()).toEqual(
      '<script type="application/json" id="amp-subscriptions">{"services":[{"authorizationUrl":"https://pub.com/amp-authorisation?rid=READER_ID&url=SOURCE_URL","pingbackUrl":"https://pub.com/amp-pingback?rid=READER_ID&url=SOURCE_URL","actions":{"login":"https://pub.com/amp-login?rid=READER_ID&url=SOURCE_URL","subscribe":"https://pub.com/amp-subscribe?rid=READER_ID&url=SOURCE_URL"}}],"score":{"supportsViewer":10,"isReadyToPay":9}}</script>'
    );
  });
  it("should render Subscription Fallback Entitlement ", () => {
    const wrapper = shallow(<Subscription services={noServices} fallbackEntitlement={fallbackEntitlement} />);
    expect(wrapper.find("script[type='application/json']").html()).toEqual(
      '<script type="application/json" id="amp-subscriptions">{"services":[{"authorizationUrl":"","pingbackUrl":"","actions":{"login":"","subscribe":""}}],"fallbackEntitlement":{"source":"fallback","granted":true,"grantReason":"SUBSCRIBER","isLoggedIn":false}}</script>'
    );
  });
});
