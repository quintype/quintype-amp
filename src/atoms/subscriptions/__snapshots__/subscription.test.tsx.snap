// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subscriptions should render Subscription services  1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <script
      custom-element="amp-subscriptions"
      src="https://cdn.ampproject.org/v0/amp-subscriptions-0.1.js"
    />
  </HelmetWrapper>
  <script
    dangerouslySetInnerHTML={
      Object {
        "__html": "{\\"services\\":[{\\"authorizationUrl\\":\\"https://pub.com/amp-authorisation?rid=READER_ID&url=SOURCE_URL\\",\\"pingbackUrl\\":\\"https://pub.com/amp-pingback?rid=READER_ID&url=SOURCE_URL\\",\\"actions\\":{\\"login\\":\\"https://pub.com/amp-login?rid=READER_ID&url=SOURCE_URL\\",\\"subscribe\\":\\"https://pub.com/amp-subscribe?rid=READER_ID&url=SOURCE_URL\\"}}]}",
      }
    }
    id="amp-subscriptions"
    type="application/json"
  />
</Fragment>
`;
